activesupport/test/core_ext/duration_test.rb

```diff
@@ -735,6 +735,17 @@ def test_non_numeric_build_raises_error
     assert_equal "can't build an ActiveSupport::Duration from a NilClass", error.message
   end
 
+  def test_duration_symmetry
+    time = Time.parse("Dec 7, 2021")
+    expected_time = time + -1.second
+
+    assert_equal -1.second, ActiveSupport::Duration.build(-1)
+    assert_equal expected_time, time + ActiveSupport::Duration.build(1) * -1
+    assert_equal expected_time, time + -ActiveSupport::Duration.build(1)
+    assert_equal expected_time, time + ActiveSupport::Duration::Scalar.new(-1)
+    assert_equal expected_time, time + ActiveSupport::Duration.build(-1)
```

Comment:
This does not behave as expected, prior to the patch in `ActiveSupport::Duration.build`